File: window/puzzlewindow.py
from window.template import OpaqueWindow

class PuzzleWindow(OpaqueWindow):
  def __init__(self, engine, **kwargs):
    super().__init__(engine, 'qml/Puzzle.qml', **kwargs)
    self.Global.puzzleWindowOpenChanged.connect(self.onWindowOpenChanged)

  def onWindowOpenChanged(self):
    isOpen = self.Global.property('puzzleWindowOpen')
    self.setVisible(isOpen)
    self.rootObject().setProperty('switchModel', 0)
    self.rootObject().setProperty('guideModel', 0)

File: app/src/window/puzzlewindow.cpp
#include "puzzlewindow.h"

PuzzleWindow::PuzzleWindow(QQmlEngine* engine)
    : Window(engine, false)
{
    setSource(QUrl("qrc:/ui/Puzzle.qml"));

    connect(global(), SIGNAL(puzzleWindowOpenChanged()),
        this, SLOT(onWindowOpenChanged()));
}

void PuzzleWindow::onWindowOpenChanged()
{
    bool open = global()->property("puzzleWindowOpen").toBool();
    setVisible(open);
    rootObject()->setProperty("switchModel", 0);
    rootObject()->setProperty("guideModel", 0);
}
